auth/auth.go

```diff
@@ -10,12 +10,36 @@ import (
 	"log/slog"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"golang.org/x/crypto/ssh"
 )
 
 const defaultPrivateKey = "id_ed25519"
 
+func keyPath() (string, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Join(home, ".ollama", defaultPrivateKey), nil
+}
+
+func GetPublicKey() (string, error) {
+	keyPath, err := keyPath()
+	if err != nil {
+		return "", err
+	}
+
+	pubKey, err := os.ReadFile(keyPath + ".pub") // note: the .pub suffix is extremely important to not leak the private key
```

Comment:
Can we parse the pubkey to _ensure_ that it is actually a public key? I think we can use `ssh.ParseAuthorizedKey([]byte(pubKey))` from `crypto/ssh`. That would make it so we _always_ return a pubkey instead of something else.

Reply:
Good call, done